### ISSUE TITLE
Artefacts When Using StableDiffusionControlNetInpaintPipeline with Multiple ControlNets

### ISSUE THREAD

**Issue by vadsuh** [huggingface/diffusers#5558]
### Describe the bug

I'm encountering an issue with the **StableDiffusionControlNetInpaintPipeline** when I use multiple ControlNet models, specifically the inpaint with openpose model. The images I'm getting have unexpected openpose patterns popping up. I'm seeing shapes and patterns that I didn't anticipate.

| Input Image | Mask |
|---------|---------|
| ![yoga](https://github.com/huggingface/diffusers/assets/34363498/b4d2275e-dba9-4fcd-92fa-771c6443ff89) |![yoga-mask](https://github.com/huggingface/diffusers/assets/34363498/666dc6ac-43a0-49b8-9db2-8b1bc183ff0d)|

| Output  |
|---------|
|![download](https://github.com/huggingface/diffusers/assets/34363498/67da1772-beca-44e3-8150-8ac9e7122f8f) |

### Reproduction

```
import numpy as np
import torch
from PIL import Image

from diffusers import ControlNetModel, StableDiffusionControlNetInpaintPipeline, EulerAncestralDiscreteScheduler
from diffusers.utils import load_image
from controlnet_aux import OpenposeDetector


def make_inpaint_condition(image, image_mask):
    image = np.array(image.convert("RGB")).astype(np.float32) / 255.0
    image_mask = np.array(image_mask.convert("L")).astype(np.float32) / 255.0

    assert image.shape[0:1] == image_mask.shape[0:1], "image and image_mask must have the same image size"
    
    image[image_mask > 0.5] = -1.0  # set as masked pixel
    image = np.expand_dims(image, 0).transpose(0, 3, 1, 2)
    image = torch.from_numpy(image)
    
    return image


def make_openpose_image(image: Image.Image) -> Image.Image:
    open_pose = OpenposeDetector.from_pretrained("lllyasviel/Annotators")
    return open_pose(image, hand_and_face=False)


# Load models
inpaint_model_path = "lllyasviel/control_v11p_sd15_inpaint"
openpose_model_path = "lllyasviel/control_v11p_sd15_openpose"

controlnet_inpaint_model = ControlNetModel.from_pretrained(inpaint_model_path, torch_dtype=torch.float16)
controlnet_openpose_model = ControlNetModel.from_pretrained(openpose_model_path, torch_dtype=torch.float16)

controlnets = [
    controlnet_inpaint_model,
    controlnet_openpose_model
]

base_model_path = "runwayml/stable-diffusion-v1-5"
controlnet_inpaint_pipeline = StableDiffusionControlNetInpaintPipeline.from_pretrained(
    base_model_path, controlnet=controlnets, safety_checker=None, torch_dtype=torch.float16
)
controlnet_inpaint_pipeline.scheduler = EulerAncestralDiscreteScheduler.from_config(controlnet_inpaint_pipeline.scheduler.config)
controlnet_inpaint_pipeline.enable_model_cpu_offload()

# Load images
init_image = load_image("/workspace/init-image.jpg")
mask_image = load_image("/workspace/yoga-mask.jpg")

# Create control images
openpose_control_image = make_openpose_image(init_image)
inpaint_control_image = make_inpaint_condition(init_image, mask_image)

# Inpainting process
generator = torch.Generator(device="cpu")
generator.manual_seed(1)

output_image = controlnet_inpaint_pipeline(
    prompt="super-hero character, best quality, extremely detailed",
    negative_prompt="monochrome, lowres, bad anatomy, worst quality, low quality",
    num_inference_steps=40,
    generator=generator,
    image=[init_image],
    mask_image=mask_image,
    control_image=[inpaint_control_image, openpose_control_image],
    controlnet_conditioning_scale=[0.5, 0.5]
).images[0]

display(output_image)
```

### System Info

- `diffusers` version: 0.22.0.dev0
- Platform: Linux-5.4.0-148-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version (GPU?): 2.0.1+cu118 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.30.2
- Accelerate version: 0.21.0
- xFormers version: 0.0.21

### Who can help?

 @sayakpaul @yiyixuxu @DN6 @patrickvonplaten

**Comment by DN6** [comment]
@vadsuh Are you seeing the same effect when increasing the conditioning scale of the openpose model? e.g. to 1.0?

**Comment by vadsuh** [comment]
@DN6 Yes, I see the same effect even when increasing the conditioning scale of the openpose model to 1.0.
![download-1](https://github.com/huggingface/diffusers/assets/34363498/4d01cf21-2fde-484c-bc38-66a47d845272)

**Comment by sayakpaul** [comment]
Does the regular ControlNet pipeline work okay when combined with the same ControlNets you're using here? That would help us understand better if the inpainting pipeline is flawed or not. @yiyixuxu WDYT?
